### ISSUE TITLE
map[x]Thing is in an invalid state from my_map['missing_entry']

### ISSUE THREAD

**Issue by donpdonp** [vlang/v#9860]
It looks like a struct Thing is only half-initialized when it comes from a map[T]Thing. The thing  instance's name field works okay, but accessing the bob instance's name field is a runtime error.

```v
struct Thing {
	name string
}

fn main() {
	thing := Thing{}
	println('thing is $thing')
	assert thing.name != 'bob'

	names := map[string]Thing{}
	bob := names['bob'] // non-existant key returns empty struct
	println('bob is $bob')
	// OK
	assert bob.name == ''

	// BOOM
	assert bob.name != 'bob'
}
```

```bash
$ v run map_get.v
thing is Thing{
    name: ''
}
bob is Thing{
    name: ''
}
V panic: string.eq(): nil string
v hash: 6e589e2
/tmp/v/map_get.10395632987062400370.tmp.c:5676: at v_panic: Backtrace
/tmp/v/map_get.10395632987062400370.tmp.c:7984: by string_eq
/tmp/v/map_get.10395632987062400370.tmp.c:7996: by string_ne
/tmp/v/map_get.10395632987062400370.tmp.c:9453: by main__main
/tmp/v/map_get.10395632987062400370.tmp.c:9551: by main
```

**Comment by donpdonp** [comment]
here are the two different initializations:
```c 
       main__Thing thing = (main__Thing){.name = (string){.str=(byteptr)"", .is_lit=1},};
```
this one looks a lot more 'empty'. the third parameter to map_get is returned when the key is not found. 
```c
        main__Thing bob = (*(main__Thing*)map_get(ADDR(map, names), &(string[]){_SLIT("bob")}, &(main__Thing[]){ {0} }));
 ```

**Comment by dumblob** [comment]
Probably a typo:
```v
	// OK
	assert bob.name == ''

	// BOOM
	assert bob.name != 'bob'
```
->
```v
	// OK
	assert thing.name == ''

	// BOOM
	assert bob.name != 'bob'
```

**Comment by spytheman** [comment]
Fixed by aebb551, it now prints:
```
thing is Thing{
    name: ''
}
bob is Thing{
    name: ''
}
```
